Lib/sqlite3/test/dbapi.py

```diff
@@ -181,6 +182,11 @@ def __fspath__(self):
         with sqlite.connect(path) as cx:
             cx.execute('create table test(id integer)')
 
+        # bpo-42862: addCleanup fails to unlink TESTFN on Windows unless we
+        # explicitly close the connection and run the GC.
+        cx.close()
+        gc_collect()
```

Comment:
As @vstinner I think this is wrong: if this is really needed, is hiding a real bug

Reply:
Yes, I agree. Let's try to fix this in #26462 and then rebase this PR onto `main` when it is resolved. For now, I'll revert the workaround.